silkworm/node/stagedsync/execution_engine.cpp

```diff
@@ -64,7 +64,7 @@ void ExecutionEngine::insert_blocks(const std::vector<std::shared_ptr<Block>>& b
     }
 }
 
-bool ExecutionEngine::insert_block(const std::shared_ptr<Block> block) {
+bool ExecutionEngine::insert_block(const std::shared_ptr<Block>& block) {
```

Comment:
I don't always agree with clang-tidy's suggestions. For example here clang-tidy suggests us to use const& but I consider passing shared_ptr by reference a wrong pattern: shared_ptr is built to be copied, otherwise we need to use it carefully, for example paying attention to not to pass the reference to shared_ptr to a closure that than is pushed to a thread.

Reply:
There's a nice article here by Herb Sutter about smart pointer parameter passing https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/

Reply:
I know, thanks. Not many agree with those rules because they clash with past conventions. In any case, the point is another: is a rule encoded in clang-tidy always valid, regardless of the context, so that we can rely on clang-tidy to ensure that it is respected?

Reply:
I think it's fine to follow Herb's advise since we don't have any legacy code in Silkworm and thus past conventions aren't relevant to us.

clang-tidy isn't perfect, but it can be tuned to be of net benefit. We can disable any problematic check altogether in `.clang-tidy` or at a specific code site with `// NOLINT`. To my mind we should [use the tools available](https://github.com/cpp-best-practices/cppbestpractices/blob/master/02-Use_the_Tools_Available.md). C++ is a complicated beast, so we should use all the help we can get to rein it in :)